### PR TITLE
[vectorized](udf) java udf support map type

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

Currently, returning a value of type 'map' is not supported.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 43.49 seconds
 stream load tsv:          502 seconds loaded 74807831229 Bytes, about 142 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          28.9 seconds inserted 10000000 Rows, about 346K ops/s
 storage size: 17169689036 Bytes

File: fe/be-java-extensions/java-udf/src/main/java/org/apache/doris/udf/UdfExecutor.java [hunk 1]
```diff
@@ -503,4 +573,78 @@ protected void init(TJavaUdfExecutorCtorParams request, String jarPath, Type fun
...
+        public Object[] get(Object[] keyCol, Object[] valueCol) {
+            Object[] retHashMap = new HashMap[keyCol.length];
+            for (int colIdx = 0; colIdx < keyCol.length; colIdx++) {
+                HashMap<T1, T2> hashMap = new HashMap<>();
+                ArrayList<T1> keys = (ArrayList<T1>) (keyCol[colIdx]);
+                ArrayList<T2> values = (ArrayList<T2>) (valueCol[colIdx]);
+                for (int i = 0; i < keys.size(); i++) {
+                    T1 key = keys.get(i);
+                    T2 value = values.get(i);
+                    hashMap.put(key, value);
```
Comment (HappenLee) [inline fe/be-java-extensions/java-udf/src/main/java/org/apache/doris/udf/UdfExecutor.java:643]: keep the same as doris map if contain the key， just skip the put call continue of call put_if_not_exist

File: fe/be-java-extensions/java-udf/src/main/java/org/apache/doris/udf/BaseExecutor.java [hunk 2]
```diff
@@ -1202,4 +1202,238 @@ public Object[] convertArrayArg(int argIdx, boolean isNullable, int rowStart, in
         }
         return argument;
     }
+
+    public Object[] convertMapKeyArg(int argIdx, boolean isNullable, int rowStart, int rowEnd, long nullMapAddr,
```
Comment (HappenLee) [inline fe/be-java-extensions/java-udf/src/main/java/org/apache/doris/udf/BaseExecutor.java:1206]: `convertMapKeyArg` look same as `convertMapValueArg` maybe we only need keep one?

**Comment by Mryange** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 44.47 seconds
 stream load tsv:          509 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17160604732 Bytes

**Comment by Mryange** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 44.4 seconds
 stream load tsv:          504 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         31 seconds loaded 2358488459 Bytes, about 72 MB/s
 stream load orc:          71 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17167881895 Bytes

**Comment by Mryange** [comment]
run arm

**Review by zhangstar333 — APPROVED** [review]

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**